sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -114,8 +110,32 @@
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ConfigurationSetting>> addSetting(ConfigurationSetting setting) {
-        return monoContext(context -> addSetting(setting, context));
+    public Mono<ConfigurationSetting> addSetting(ConfigurationSetting setting) {
+        return monoContext(context -> addSetting(setting, context))
+            .flatMap(response -> Mono.justOrEmpty(response.value()));
+    }
+
+    /**
+     * Adds a configuration value in the service if that key does not exist.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Add a setting with the key "prodDBConnection" and value "db_connection".</p>
+     *
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.addSettingWithResponse#string-string}
+     *
+     * @param key The key of the configuration setting to add.
+     * @param value The value associated with this configuration setting key.
+     * @return The {@link ConfigurationSetting} that was created with the REST response, or {@code null}, if a key collision occurs or the key
+     * is an invalid value (which will also throw HttpResponseException described below).
+     * @throws IllegalArgumentException If {@code key} is {@code null}.
+     * @throws ResourceModifiedException If a ConfigurationSetting with the same key exists.
+     * @throws HttpResponseException If {@code key} is an empty string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<ConfigurationSetting>> addSettingWithResponse(String key, String value) {
+        return monoContext(
+            context -> addSetting(new ConfigurationSetting().key(key).value(value), context));
```

Comment:
I don't think this overload is necessary - users can use the other addSettingWithResponse to achieve the same thing.